src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -118,19 +102,20 @@ public bool TryGoToDefinition(Document document, int position, CancellationToken
                 FindUsagesHelpers.GetDisplayName(symbol));
 
             return _threadingContext.JoinableTaskFactory.Run(() =>
-                _streamingPresenter.Value.TryNavigateToOrPresentItemsAsync(
+                _streamingPresenter.TryNavigateToOrPresentItemsAsync(
                     _threadingContext, solution.Workspace, title, definitions));
         }
 
         private static bool IsThirdPartyNavigationAllowed(ISymbol symbolToNavigateTo, int caretPosition, Document document, CancellationToken cancellationToken)
         {
             var syntaxRoot = document.GetSyntaxRootSynchronously(cancellationToken);
-            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+            var syntaxFactsService = document.GetRequiredLanguageService<ISyntaxFactsService>();
             var containingTypeDeclaration = syntaxFactsService.GetContainingTypeDeclaration(syntaxRoot, caretPosition);
 
             if (containingTypeDeclaration != null)
             {
                 var semanticModel = document.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                Debug.Assert(semanticModel != null);
```

Comment:
why not just make the above GetRequiredSemnaticModelAsync?

Reply:
Because that returns a `ValueTask`, which does not have `WaitAndGetResult` defined on it. I didn't want to make any logic changes beyond nullable things in this PR, so I'll leave it as is.